# whether the robot is in the pipeline or not
bool in
# the ID of the current segment in which the robot is
int32 id
# the previous and next ID from current ID
int32 previd
int32 nextid
# the distance from start position and end position in the current pipe segment
float64 startdist
float64 enddist
